atc/db/team.go

```diff
@@ -647,6 +677,55 @@ func (t *team) CreateOneOffBuild() (Build, error) {
 	return build, nil
 }
 
+func (t *team) CreateStartedBuild(plan atc.Plan) (Build, error) {
+	tx, err := t.conn.Begin()
+	if err != nil {
+		return nil, err
+	}
+
+	defer Rollback(tx)
+
+	metadata, err := json.Marshal(plan)
+	if err != nil {
+		return nil, err
+	}
+
+	encryptedPlan, nonce, err := t.conn.EncryptionStrategy().Encrypt(metadata)
+	if err != nil {
+		return nil, err
+	}
+
+	build := &build{conn: t.conn, lockFactory: t.lockFactory}
+	err = createBuild(tx, build, map[string]interface{}{
+		"name":         sq.Expr("nextval('one_off_name')"),
+		"team_id":      t.id,
+		"status":       BuildStatusStarted,
+		"start_time":   sq.Expr("now()"),
+		"schema":       "exec.v2",
+		"private_plan": encryptedPlan,
+		"public_plan":  plan.Public(),
+		"nonce":        nonce,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	err = build.saveEvent(tx, event.Status{
```

Comment:
same concern as above - i'm a little iffy on inserting these in 'started' state

Reply:
RIP BuildStarter, long live BuildPlanner